### PR TITLE
Add type definitions for all classes in the shapes namespace for Join…

### PR DESCRIPTION
**Pull request by ChrisMoran**
…tJS.

Please fill in this template.

- [ X] Make your PR against the `master` branch.
- [ X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ X] Test the change in your own code. (Compile and run.)
- [X ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ X] Run `tsc` without errors.
- [X ] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [X ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/clientIO/joint/tree/master/plugins/shapes
- [ X] Increase the version number in the header if appropriate.
- [X ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by CaselIT** [comment]
Serena ok. :+1:

File: jointjs/index.d.ts [hunk 1]
```diff
@@ -492,13 +501,285 @@ declare namespace joint {
...
+
+        namespace devs {
+            interface ModelAttributes extends GenericAttributes<dia.SVGAttributes> {
+                inPorts?: string[];
+                outPorts?: string[];
+                ports?: Object;
+            }
+            class Model extends basic.Generic {
+                constructor(attributes?: ModelAttributes, options?: Object);
+                changeInGroup(properties: any, opt?: Object): boolean;
```
Comment (zhengbli) [inline jointjs/index.d.ts]: Don't use `Object`, use `any` or `{}` instead. http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

File: jointjs/index.d.ts [hunk 2]
```diff
@@ -492,13 +501,285 @@ declare namespace joint {
...
+            class Multivalued extends Attribute {
+            }
+            class Derived extends Attribute {
+            }
+            class Key extends Attribute {
+            }
+            class Normal extends Attribute {
+            }
+            interface ISAAttrs extends dia.Element {
+                polygon?: ShapeAttrs
```
Comment (zhengbli) [inline jointjs/index.d.ts]: semicolon

File: jointjs/index.d.ts [hunk 3]
```diff
@@ -492,13 +501,285 @@ declare namespace joint {
...
+            }
+            class ISA extends dia.Element {
+                constructor(attributes?: GenericAttributes<ISAAttrs>, options?: Object)
+            }
+            class Line extends dia.Link {
+                cardinality(value: string | number): void
```
Comment (zhengbli) [inline jointjs/index.d.ts]: semicolon

File: jointjs/index.d.ts (hunk 1 above)
Comment (ChrisMoran) [inline jointjs/index.d.ts]: Changed the options? args to any.

File: jointjs/index.d.ts (hunk 2 above)
Comment (ChrisMoran) [inline jointjs/index.d.ts]: Added semicolons to the end of all added lines.

File: jointjs/index.d.ts (hunk 3 above)
Comment (ChrisMoran) [inline jointjs/index.d.ts]: Added semicolons to the end of all added lines.

**Comment by ChrisMoran** [comment]
Addressed comments in previous version, please take another look!

**Comment by ChrisMoran** [comment]
@zhengbli Any additional comments?

**Comment by zhengbli** [comment]
👍

**Pull request merged**